### ISSUE TITLE
VSCode Very Slow Startup

### ISSUE THREAD

**Issue by mehraneslaminia** [microsoft/vscode#158620]
VS Code takes a long time to start. Please see the Startup Performance table below. Even after it is ready, opening a New Window takes a long time. I have already tried the Insider Version and disabled all extensions but still, the startup time is very slow.

## System Info

* Code: 1.70.2 (e4503b30fc78200f846c62cf8091b76ff5547662)
* OS: win32(10.0.19044)
* CPUs: AMD Ryzen Threadripper PRO 5945WX 12-Cores     (24 x 4092)
* Memory(System): 31.86 GB(23.82GB free)
* Memory(Process): 187.23 MB working set(157.21MB private, 0.09MB shared)
* VM(likelihood): 0%
* Initial Startup: true
* Has 0 other windows
* Screen Reader Active: false
* Empty Workspace: true

## Performance Marks

| What                                                            | Duration | Process                   | Info                                                       |
| --------------------------------------------------------------- | -------- | ------------------------- | ---------------------------------------------------------- |
| start => app.isReady                                            | 10074    | [main]                    | initial startup: true                                      |
| nls:start => nls:end                                            | 0        | [main]                    | initial startup: true                                      |
| require(main.bundle.js)                                         | 53       | [main]                    | initial startup: true                                      |
| start crash reporter                                            | 10029    | [main]                    | initial startup: true                                      |
| serve main IPC handle                                           | 1        | [main]                    | initial startup: true                                      |
| create window                                                   | 19       | [main]                    | initial startup: true, state: 0ms, widget: 11ms, show: 8ms |
| app.isReady => window.loadUrl()                                 | 126      | [main]                    | initial startup: true                                      |
| window.loadUrl() => begin to require(workbench.desktop.main.js) | 24689    | [main->renderer]          | NewWindow                                                  |
| require(workbench.desktop.main.js)                              | 197      | [renderer]                | cached data: YES, node_modules took 0ms                    |
| wait for window config                                          | 0        | [renderer]                | -                                                          |
| init storage (global & workspace)                               | 10       | [renderer]                | -                                                          |
| init workspace service                                          | 23       | [renderer]                | -                                                          |
| register extensions & spawn extension host                      | 125      | [renderer]                | -                                                          |
| restore viewlet                                                 | 0        | [renderer]                | -                                                          |
| restore panel                                                   | 0        | [renderer]                | -                                                          |
| restore & resolve visible editors                               | 6        | [renderer]                | 0:                                                         |
| overall workbench load                                          | 166      | [renderer]                | -                                                          |
| workbench ready                                                 | 35281    | [main->renderer]          | -                                                          |
| renderer ready                                                  | 396      | [renderer]                | -                                                          |
| shared process connection ready                                 | 0        | [renderer->sharedprocess] | -                                                          |
| extensions registered                                           | 35475    | [renderer]                | -                                                          |

It also shows the following message. I just click on Keep Waiting and VSCode fully deployed after a while. 
![Capture](https://user-images.githubusercontent.com/4812769/185687306-b6f4e9c1-6375-42be-9be4-011a195c0e42.PNG)

I have created two profile files by running `code --prof-startup`. Please see the attachments below: 
[prof-zEgx9l46.renderer.cpuprofile.txt](https://github.com/microsoft/vscode/files/9384743/prof-zEgx9l46.renderer.cpuprofile.txt)
[prof-zEgx9l46.main.cpuprofile.txt](https://github.com/microsoft/vscode/files/9384741/prof-zEgx9l46.main.cpuprofile.txt)

**Comment by jrieken** [comment]
One part is https://github.com/microsoft/vscode/issues/22087 - which is about a delayed app ready event. However, there seems to be another issue that launching new processes (like new windows) takes 20+ seconds. 

@mehraneslaminia This can be caused by AV software. You could try to exclude VS Code from it and see if that makes a difference

**Comment by deepak1556** [comment]
@mehraneslaminia can you get the trace following the steps below to help us understand why the process launches are slow on your device, thanks!

* Open `Windows Performance Recorder` from system start menu
* Have the following settings

![Screenshot 2022-08-23 200440](https://user-images.githubusercontent.com/964386/186142701-ecc01966-8cc5-4d17-aa23-d896a925f083.png)


* Start the recorder
* Open the application
* Once the editor has completely loaded, stop the recorder and save the profile
* Send the profile to Deepak.Mohan@microsoft.com instead of attaching to this issue, since it will contain sensitive information.

**Comment by mehraneslaminia** [comment]
I need to consult with our IT about items because the machine is managed by my company. I will keep you posted.

**Comment by mehraneslaminia** [comment]
On my machine with Windows 11, the issue has been resolved by the recent Windows update on October 11 [KB5018418](https://support.microsoft.com/en-us/topic/october-11-2022-kb5018418-os-build-22000-1098-62c0cb53-7562-4089-8672-5c80845932c1). However, we don't know what is the cause of the issue and how this new update has fixed it yet.

**Comment by thexperiments** [comment]
I'm on windows 10 here with all updates installed and the problem still exists.

**Comment by tusharsnn** [comment]
On my system [Windows 11 insiders beta], Vscode seems to trigger wsl before it loads the app window. Earlier this is how it used to be: 
```
vscode icon is clicked -> app window visible (content is loading) -> WSL VM visible on Task Manager -> wsl loaded -> content visible
```

where as now:
```
vscode icon is clicked -> WSL VM visible on Task Manager -> app window visible (content is loading) -> wsl loaded -> content visible
```


So the difference is app window loads only after wsl Vm is fully active which is delaying the app window to load. Also, **this happens even if no wsl window is/being opened**. So opening a file from LocalDisk also causes *delayed* app window.

If *Wsl VM* is already running, app window opens without any delay (does not matter remote/local window).

### Startup Performance Output

- #### VM is running
| What                                                            | Duration | Process                   | Info                                                        |
| --------------------------------------------------------------- | -------- | ------------------------- | ----------------------------------------------------------- |
| window.loadUrl() => begin to require(workbench.desktop.main.js) | 155      | [main->renderer]          | NewWindow                                                   |

- #### VM is not running
| What                                                            | Duration | Process                   | Info                                                        |
| --------------------------------------------------------------- | -------- | ------------------------- | ----------------------------------------------------------- |
| window.loadUrl() => begin to require(workbench.desktop.main.js) | 4323      | [main->renderer]          | NewWindow 

I am using SSD but people on HDD will see even higher app window load time due to VM's slow startup.

**Comment by tusharsnn** [comment]
**I could no longer reproduce the issue**. 
Somehow issue magically went away. I was able to reproduce the issue before that. If anybody's experiencing a similar situation just ping me, and I'll share what I did before it got fixed. Note that I'm not sure what exactly fixed it.

**Comment by deepak1556** [comment]
@thexperiments if the issue persists with `1.74`, can you take a new WPR trace and I can provide the necessary runtime symbols. The previous one attached in this issue thread is not symbolicated.

**Comment by VSCodeTriageBot** [comment]
This issue has been closed automatically because it needs more information and has not had recent activity. See also our [issue reporting](https://aka.ms/vscodeissuereporting) guidelines.

Happy Coding!
